test/cql-pytest/test_secondary_index.py

```diff
@@ -241,3 +241,23 @@ def test_filter_cluster_key(cql, test_keyspace):
         stmt = SimpleStatement(f"SELECT c1, c2 FROM {table} WHERE c1 = 1 and c2 = 1 ALLOW FILTERING")
         rows = cql.execute(stmt)
         assert_rows(rows, [1, 1])
+
+# Reproduces #9085.  Scylla-only because Cassandra uses indices differently.
+def test_multi_column_skips_regular_index(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c1 int, c2 int, r int, primary key(p,c1,c2)') as tbl:
+        cql.execute(f'CREATE INDEX ON {tbl}(r)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 1, 1, 0)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 1, 2, 1)')
+        cql.execute(f'INSERT INTO {tbl}(p, c1, c2, r) VALUES (1, 2, 1, 0)')
+        with pytest.raises(InvalidRequest, match="use ALLOW FILTERING"):
+            cql.execute(f'SELECT c1 FROM {tbl} WHERE p=1 AND (c1,c2)<(2,0) AND r=0')
+
+@pytest.mark.xfail(reason="Other restrictions are ignored when clustering key is constrained.  Issue #6200")
```

Comment:
This is a test for a **different** issue, right (this is what you said in #9085 - that #6200 is different).
So please put this test in a separate commit, with a separate commit message and adding "Refs #6200" - so we'll remember that you added a test for 6200 when we come to work on it.

Reply:
Also, please add a comment in front (or if you must, a docstring) for the test explaining what it tests. At the very minimum, add a statement that it reproduces issue #6200. Yes, you said this in the xfail message, but the xfail will eventually be removed (that's our goal :-)), but the comment explaining why this test exists should stay behind.

Reply:
> This is a test for a **different** issue, right (this is what you said in #9085 - that #6200 is different).

No, this is exactly the test you asked about before.  It was all a part of the same test until @psarna [suggested](https://github.com/scylladb/scylla/pull/9115#discussion_r679946627) I split them.

> Also, please add a comment in front (or if you must, a docstring) for the test explaining what it tests.

What would the comment say that isn't already in the test's name?

Reply:
You wrote in the commit message "Fixes #9085". Now I understand it fixes your new definition of 9085, but it does not fix the specific example which started 9085 - which is still broken and now you call it 6200. If that's the way you decided to go, please fix the issues to follow your decision: Copy the broken example to issue 6200, close 9085 as a duplicate of 6200, and open a new issue with what you are fixing here. Alternatively, I can accept that you don't bother with fixing the issues if at least you fix the code comments and commit messages to explain what's going on.
2. 

 As a reviewer (or somebody who will read this code, or commit, or issue a year from now) I have no idea what's the difference between 9085 or 6200 or which one this commit fix - unless you explain it clearly. I understand

Reply:
> > Also, please add a comment in front (or if you must, a docstring) for the test explaining what it tests.
> 
> What would the comment say that isn't already in the test's name?

First, I answered exactly that question in the next sentence in my review: *"At the very minimum, add a statement that it reproduces issue #6200. Yes, you said this in the xfail message, but the xfail will eventually be removed (that's our goal :-)), but the comment explaining why this test exists should stay behind."*

Second, are you seriously trying to suggest that from the name "test_multi_column_skips_regular_index" a year from now anybody will understand why this test exists? That it's related to issues 6200 or 9085? Understand why a "multi_column" (multi column what?) should (or shouldn't?) skip a regular index (what's a regular index?), and what happens if it does the wrong thing?
The irony is that you did write a comment of sorts in the xfail message: "Other restrictions are ignored when clustering key is constrained. Issue #6200". I think it's still too short for my taste but it's much better than nothing. The problem with xfail message is that it will be deleted when the bug is fixed - and I want to have a comment that will stay behind. We should know that a test was written for issue 6200 long after this issue is closed - this can help us in the future if we need to backport a bugfix, or if a test starts to fail again and we want to remember why this test even exists and whether the test is even testing a correct thing and why.

Reply:
> > What would the comment say that isn't already in the test's name?
> 
> First, I answered exactly that question in the next sentence in my review: _"At the very minimum, add a statement that it reproduces issue #6200.

That statement would be wrong.  This reproduces #9085.  #6200 is just a confounding factor here.

> Second, are you seriously trying to suggest that from the name "test_multi_column_skips_regular_index"

Um, that's... not the name we're discussing.

> Understand why a "multi_column" (multi column what?)

Multi-column is only used in one sense in all source and documentation I've ever seen.

Reply:
> You wrote in the commit message "Fixes #9085". Now I understand it fixes your new definition of 9085, but it does not fix the specific example which started 9085

While I disagree with the rest of your analysis, this tidbit is absolutely valid.  Thinking about how to address it, I figured out a much better fix.  Thank you. :)